NameLayerMain/src/vg/civcraft/mc/namelayer/GroupManager.java

```diff
@@ -330,25 +330,21 @@ public boolean hasAccess(Group group, UUID player, PermissionType perm) {
 			}
 		}
 		GroupPermission perms = getPermissionforGroup(group);
-		for(PlayerType rank : getRecursivePlayerTypes(group, player)) {
-			if (perms.hasPermission(rank, perm)) {
-				//player has right rank in the group itself or at least one super group
-				return true;
-			}
-		}		
+		PlayerType rank = getRecursiveHighestPlayerType(group, player); 
+		if (perms.hasPermission(rank, perm)) {
+			//player has right rank in the group itself or at least one super group
+			return true;
+		}	
 		return false;
 	}
 	
-	private List<PlayerType> getRecursivePlayerTypes(Group group, UUID player) {
-		List<PlayerType> perms = new LinkedList<PlayerType>();
+	private PlayerType getRecursiveHighestPlayerType(Group group, UUID player) {
 		PlayerType type = group.getPlayerType(player);
-		if (type != null) {
-			perms.add(type);
-		}
 		if (group.hasSuperGroup()) {
-			perms.addAll(getRecursivePlayerTypes(group.getSuperGroup(), player));
+			//only return highest rank
+			type = PlayerType.getByID(Math.max(PlayerType.getID(type), PlayerType.getID(getRecursiveHighestPlayerType(group.getSuperGroup(), player))));
 		}
-		return perms;
+		return type;
```

Comment:
Hmm yeah, this would assume that group perms make sense and are strictly hierarchic. We could enforce them being hierarchic, but I'm not sure whether that would be a regression.

Even though this leads to weird edge cases, it's clearer to the user in my opinion, because it clearly gives each member one rank. I'm not attached to this particular solution, but I think the way it is currently isn't good and should be changed.